modules/transactionpool/update.go

```diff
@@ -387,3 +388,245 @@ func (tp *TransactionPool) PurgeTransactionPool() {
 	tp.purge()
 	tp.mu.Unlock()
 }
+
+// objectLocations are used to pinpoint the exact locations of an object in a
+// transaction set so as to avoid recomputing ObjectIDs when searching for
+// objects.
+type objectLocation struct {
+	setID   TransactionSetID
+	txIndex int
+}
+
+// findObjectLocations finds the locations that reference the given object as an
+// input.
+func (tp *TransactionPool) findObjectLocations(oid ObjectID) ([]*objectLocation, bool) {
```

Comment:
This method's implementation will need to be changed along with changes to transaction set types, since the goal for the tpool is to have child sets separate from parent sets if the fees work out that way.